cmake/sycl.cmake

```diff
@@ -0,0 +1,33 @@
+# IntelSYCL for dpcpp and icpx if the config is existed and cmake reaches the requirement
+if(CMAKE_CXX_COMPILER MATCHES "dpcpp|icpx")
+    if(CMAKE_HOST_WIN32 AND CMAKE_VERSION VERSION_GREATER_EQUAL 3.25)
+        find_package(IntelSYCL QUIET)
+    elseif(CMAKE_VERSION VERSION_GREATER_EQUAL 3.20.5)
```

Comment:
I think we should require 3.20 for SYCL if that is necessary to make IntelLLVM work correctly.

Reply:
no, it is optional. if there's no config and cmake, simply adding `-fsycl` in the following code is enough currently.